File: sharedmodule/src/main/java/com/workingbit/share/domain/impl/NewBoardRequest.java
package com.workingbit.share.domain.impl;

import com.workingbit.share.common.EnumRules;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.io.Serializable;

/**
 * Created by Aleksey Popryaduhin on 15:58 10/08/2017.
 */
@Data
@AllArgsConstructor
public class NewBoardRequest implements Serializable {
  private boolean fillBoard = true;
  private boolean black;
  private EnumRules rules;
  private Integer squareSize;
}

File: coremodule/src/main/java/com/workingbit/coremodule/domain/impl/NewBoardRequest.kt
package com.workingbit.share.domain.impl;

import com.workingbit.coremodule.common.EnumRules

/**
 * Created by Aleksey Popryaduhin on 15:58 10/08/2017.
 * TODO(MOVE TO ACTUAL CLASS)
 */
data class NewBoardRequest(private val map: HashMap<String, Any?>) {

    val fillBoard: Boolean by map
    val black: Boolean by map
    val rules: EnumRules by map
    val squareSize: Int by map
}
